src/less/mixins/carousel/base/carousel-mixins.less

```diff
@@ -26,13 +26,15 @@
     // progressive enhancement for browsers with support
     // this cannot be combined with the next rule as
     // IE/Edge will discard the entire rule
+    &:focus,
     .carousel:focus-within & {
         .show-control();
     }
 
-    .carousel:hover &,
-    &:focus {
-        .show-control();
+    @media not all and (hover: none) {
```

Comment:
Do you need `not all`? Doesn't `@media (hover: none) {` work? We are using that elsewhere. Otherwise this looks good to me.